internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -423,6 +421,240 @@ func Test_ValidateClusterClassVariable(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "pass on variable with a default that is valid by the given schema",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type:      "string",
+						MaxLength: pointer.Int64(6),
+						Default:   &apiextensionsv1.JSON{Raw: []byte(`"short"`)},
+					},
+				},
+			},
+		},
+		{
+			name: "fail on variable with a default that is invalid by the given schema",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type:      "string",
+						MaxLength: pointer.Int64(6),
+						Default:   &apiextensionsv1.JSON{Raw: []byte(`"veryLongValueIsInvalid"`)},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "pass if variable is an object with default value valid by the given schema",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
```

Comment:
Cam we also add a default value on the top-level pls? (would be nice to also have a case where the top-level default value (which is a object) does not match the schema

Reply:
Similar for example and enum (enum can be objects too). Thank you!!!